### ISSUE TITLE
Multiple Ignore Patterns

### ISSUE THREAD

**Issue by rypete** [antoine-coulon/skott#79]
I have a few different patterns I want to ignore via `--ignorePatterns` but it seems this can only accept one pattern? It would be great if this could accept an array similar to `--fileExtensions`

**Comment by jakebiesinger** [comment]
I also happened to need this today... it's hacky, but you skott is using https://www.npmjs.com/package/minimatch under the hood, so you can use `{}` with comma-separated lists, e.g.,

```
--ignorePattern '{src/file1.ts,src/path/to/file2.ts,src/api/**/*.ts}'
```

Note the surrounding `'` since your shell may try to interpolate otherwise.

**Comment by rypete** [comment]
The above are very helpful, thanks. What I ended up doing was temporarily putting the paths I wanted to ignore it my .gitignore file before running skott.

**Comment by antoine-coulon** [comment]
Hello folks, thanks for opening that issue @rypete! 

"ignorePattern" is a pretty new feature for skott so I wanted to collect few feedbacks about the ability of ignoring specific set of files before going further as there is currently the "ignorePattern" option, the "fileExtensions" option that can be also considered as a filter even though it does not really allow fine grained file filtering and also the ".gitignore" support.

If you guys are facing some use cases where you would want multiple patterns to be provided at once I can definitely work on that to make skott support it.

Are you both @jakebiesinger @rypete using skott via the API? For the CLI it would probably require a skott config file as I'm afraid for the CLI itself it would be too verbose and error prone to allow multiple ignore patterns.

FYI I'm currently on vacation till the end of the next week so don't expect quick replies until there but I will catch up asap :)

**Comment by jakebiesinger** [comment]
I'm using skott via the CLI and would love it if there were a config file I could set and forget.

For the large typescript codebase I'm working in now, we prevent PRs containing circular imports everywhere except in a set of "context" files, where we set up helper methods that dynamically import at method call time via `await import ('./some/file').doSomeThing()`.

Skott has found literally hundreds of circular imports that madge missed due to some shortcomings in how they're parsing extended tsconfig files (https://github.com/pahen/madge/issues/322) so we're cutting over to use skott everywhere.

**Comment by rypete** [comment]
I'm using the CLI but not the config file. My use case is a lot simpler than @jakebiesinger and I only needed it for a showcase demo.

**Comment by antoine-coulon** [comment]
Thanks for the input, by the way the config file is not yet available, we were just discussing about the possibility of implementing it.

Without that config file I hardly see how we could easily provide multiple ignore patterns using the CLI

**Comment by rypete** [comment]
One way might be, and what I also tried, was to use the `--ignorePatterns` flag multiple times. I noticed the underlying system turned it into an array. This might be useful but I will of course use whatever you come up with.

Something like `skott --ignorePatterns="**/*.test.ts" --ignorePatterns="**/*.js"` etc

**Comment by antoine-coulon** [comment]
Could actually be one way to do it, didn't think about that actually, thanks!

I'm back from vacation so you'll get updates pretty soon, stay tuned!

**Comment by eppisapiafsl** [comment]
You need to use brackets `{ }` to ignore multiple files

This worked for me 
```
--ignorePattern='{commons/Store/{Redux/types/StateMachineContext.ts,ExerciseTherapySessionState/actions/transition.ts},commons/Utils/TestMocks/Provider.tsx}'
```

**Comment by antoine-coulon** [comment]
Thanks for the tip @eppisapiafsl! This is what can be used until I investigate in providing a way to stack multiple patterns without losing too much DX

**Comment by antoine-coulon** [comment]
Hello guys,

I was finally able to land support for multiple ignore patterns. It will be released in the next version, that is `skott@0.35.0`.

Multiple patterns are supported through the CLI by having multiple occurrences of the `--ignorePattern` parameter or supported within the API using a simple array `skott({ ignorePatterns: ["test/**/*.js", "examples/**/*.js" })`

Feel free to re-open the issue if you're still encountering issues after the release
